test/unit/package_item_test.rb

```diff
@@ -0,0 +1,237 @@
+    assert_equal 1, PackageItem.new(@name, @weight, @value, 0).quantity
+  end
+
+  def test_initialize_weight_mass_object
+    begin
+      @item = PackageItem.new(@name, @mass, @value, @quantity, @options)
+      assert_equal @mass, @item.weight
+      flunk "This code path is broken but passed unexpectedly"
+    rescue NameError
+      skip "This code path is broken"
+    end
+  end
+
+  def test_initialize_weight_default_metric
+    assert_equal @weight, @item.grams
+    refute_equal @weight, @item.ounces
+  end
+
+  def test_initialize_weight_accepts_imperial
+    @item = PackageItem.new(@name, @weight, @value, @quantity, @options.merge(units: :imperial))
+
+    assert_equal @weight, @item.ounces
+    refute_equal @weight, @item.grams
+  end
+
+  def test_initialize_weight_accepts_metric
+    @item = PackageItem.new(@name, @weight, @value, @quantity, @options.merge(units: :metric))
+
+    assert_equal @weight, @item.grams
+    refute_equal @weight, @item.ounces
+  end
+
+  def test_initialize_weight_does_not_accept_strings
+    @item = PackageItem.new(@name, @weight, @value, @quantity, @options.merge(units: "imperial"))
+
+    assert_equal @weight, @item.grams
+    refute_equal @weight, @item.ounces
+  end
+
+  def test_initialize_value_from_cents
+    @item = PackageItem.new(@name, @weight, "1.23", @quantity, @options)
+
+    assert_equal 123, @item.value
+  end
+
+  def test_weight
+    assert_equal @mass, @item.weight
+    assert_instance_of ::Quantified::Mass, @item.weight
+  end
+
+  def test_weight_actual
+    assert_equal @mass, @item.weight(type: :actual)
+    assert_instance_of ::Quantified::Mass, @item.weight(type: :actual)
+  end
+
+  def test_weight_volumetric
+    begin
+      assert_equal :todo, @item.weight(type: :volumetric)
+      assert_instance_of ::Quantified::Mass, @item.weight(type: :volumetric)
+      flunk "This code path is broken but passed unexpectedly"
+    rescue NoMethodError
+      skip "This code path is broken"
+    end
+  end
+
+  def test_weight_dimensional
+    begin
+      assert_equal :todo, @item.weight(type: :dimensional)
+      assert_instance_of ::Quantified::Mass, @item.weight(type: :dimensional)
+      flunk "This code path is broken but passed unexpectedly"
+    rescue NoMethodError
+      skip "This code path is broken"
+    end
+  end
+
+  def test_weight_billable_max_weight_and_volumetric
+    begin
+      assert_equal :todo, @item.weight(type: :billable)
+      assert_instance_of ::Quantified::Mass, @item.weight(type: :billable)
+      flunk "This code path is broken but passed unexpectedly"
+    rescue NoMethodError
+      skip "This code path is broken"
+    end
+  end
+
+  def test_grams_value
+    assert_equal 100, @item.grams
+  end
+
+  def test_grams_accepts_options_with_type
+    begin
+      assert_equal :todo, @item.grams(type: :volumetric)
+      flunk "This code path is broken but passed unexpectedly"
+    rescue NoMethodError
+      skip "This code path is broken"
+    end
```

Comment:
A couple of things:

1. Although this is slightly different, I wonder would `assert_raises(NameError) { ... }` basically do the same thing for us?
2. If we want this exactly, it's used enough that maybe a helper function in `TestHelper` would be nice? `broken_with(NameError) { ... }`, or whatever name you please. I'm fine keeping these here too, since they'll hopefully disappear soon!